### ISSUE TITLE
non-default POTCAR causing total NELECT to be incorrect

### ISSUE THREAD

**Issue by joebesity** [kavanase/doped#1]
Recent change to select which POTCAR is generated for each element when writing vasp_gam, vasp_std files using the argument:

`potcar_settings={"POTCAR": {"Sn": "Sn"}}
`

when the default POTCAR for Sn is Sn_d. However, the total NELECT generated in the INCAR is calculated from the valence electrons from the default POTCAR, while the updated POTCAR is copied to POTCAR file. So I think the POTCAR implementation needs to come before the determination of NELECT.

I confirmed this by running a simple bash script to calculate the number of electrons in the system from POSCAR and POTCAR:

```
species+=($(head -6 POSCAR | tail -1))
numbers+=($(head -7 POSCAR | tail -1))
electrons+=($(grep ZVAL POTCAR | awk '{print $6/1}'))
list=$(echo ${#numbers[@]} -1 | bc)

for i in $(seq 0 $list); do
       sum[$i]+=$(echo "${numbers[$i]} * ${electrons[$i]}" | bc)
done

for j in ${sum[@]}; do
       let tot+=$j
done
echo "electrons = ${tot}"
```

which returns a different value (by a couple hundred electrons...) to the value of NELECT that `doped` writes to the INCAR.

I noticed this was an issue because all of my calculations (at gamma point) were returning hugely positive energies!!

Cheers

**Comment by kavanase** [comment]
Thanks very much for this @joebesity! Fixed now with commit b1fc254
This is the Jupyter notebook I used to test, if of interest or use.
[Doped_Tests.ipynb.zip](https://github.com/kavanase/doped/files/6083458/Doped_Tests.ipynb.zip)

**Comment by joebesity** [comment]
Nice on @kavanase , just been through the notebooks and looks good! Will test on my system this afternoon. Cheers!
